mesonbuild/mtest.py

```diff
@@ -1456,6 +1456,11 @@ def _get_test_cmd(self) -> T.Optional[T.List[str]]:
                            'found. Please check the command and/or add it to PATH.')
                     raise TestException(msg.format(self.test.exe_wrapper.name))
                 return self.test.exe_wrapper.get_command() + self.test.fname
+        elif self.test.cmd_is_built and is_windows():
+            test_cmd = ExternalProgram._shebang_to_cmd(self.test.fname[0])
+            if test_cmd is not None:
+                test_cmd += self.test.fname[1:]
+            return test_cmd
```

Comment:
How do we handle File and ExternalProgram here? This feels like we are not doing something earlier that we should have done...

Reply:
Ugh, makes sense.